subsystems/win32/csrsrv/api.c

```diff
@@ -645,9 +646,11 @@ CsrApiRequestThread(IN PVOID Parameter)
                     /* Remove our extra reference */
                     CsrLockedDereferenceThread(CsrThread);
```

Comment:
The CsrLocked*** functions suppose the corresponding CSR process lock is already been held by the caller.

Reply:
Nota: Both `CsrLockedDereferenceProcess()` and `CsrLockedDereferenceThread()` functions do their CsrAcquireProcessLock() misplaced (it should come after the CsrThreadRefcountZero() call).
Guilty commit is 0444a3a392837cefe4d07bcc8713bfd9b2653229.